src/AppBundle/Service/SearchUserFormHelper.php

```diff
@@ -131,9 +137,9 @@ public function getSearchForm($formBuilder, $params_string = '', $ambassador = f
     public function initSearchQuery($doctrineManager) {
         /** @var QueryBuilder $qb */
         $qb = $doctrineManager->getRepository("AppBundle:Membership")->createQueryBuilder('o');
-        $qb = $qb->leftJoin("o.beneficiaries", "b")->addSelect("b")
-            ->leftJoin("o.registrations", "r")->addSelect("r")
-            ->leftJoin("b.user", "u")->addSelect("u");
```

Comment:
Pourquoi tu supprimes les `addSelect("b")` and `addSelect("r")` ?

Je veux dire par là qu'il faudrait regarder l'impact que ça a au niveau des requetes. Si jamais l'information est utilisée dans le twig, doctrine va devroit refaire une requete part utilisateur pour aller refetcher cette info. C'est pas ouf en terme de performance.

Reply:
en effet j'aurais dû laisser un commentaire.
- oui ca a un impact sur le nombre de requêtes qu'effectue doctrine
- mais c'est le seul moyen pour garder l'affichage de tous les bénéficiaires d'un compte si l'on filtre sur l'un des deux bénéficiaires (par exemple en utilisant les filtres username / firstname / lastname / email)

le use case étant : si je cherche Bob, je veux renvoyer l'ensemble du membre dont fait partie Bob. là actuellement ca affiche Bob comme si il était seul sur son compte.

Reply:
Y'aurais pas moyen avec un `if` de rajouter les `addSelect` ? (c'est mon côté un peu maniaque d'éviter de générer des problèmes à n+1 queries)

Reply:
il faudrait ajouter les `addSelect` si aucun filtre relatifs aux modèles `Beneficiary` ou `User` sont utilisés. je ne vois pas trop comment facilement ^^
mais tu peux rajouter un commit par dessus cette PR si tu trouves

Reply:
En vrai, j'ai le souvenir que cette partie du code n'était pas terrible, je pense qu'il faudrait peut-être couvrir ce point dans un autre travail avec un refactoring.

Reply:
ca marche